api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.iceberg;
+
+import java.util.Collection;
+import java.util.concurrent.ExecutorService;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.io.CloseableIterable;
+
+/**
+ * Scan objects are immutable and can be shared between threads. Refinement methods, like
+ * {@link #select(Collection)} and {@link #filter(Expression)}, create new TableScan instances.
+ */
+interface Scan<T extends Scan> {
```

Comment:
Maybe it's better to define this `Scan` as `interface Scan<T>`, and add a javadoc saying `<T> the concrete Java class of a Scan instance.`. ( I commented this because it seems to be wired to introduce a IDEA warning saying `Raw use of parameterized class 'Scan' `)

There is a good example here:
https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/api/src/main/java/org/apache/iceberg/ContentFile.java#L29

Reply:
Then how about `Scan<T extends Scan<T>>`?

Reply:
Will try @yittg 's suggestion. I was copying the builder inheritance pattern. It wasn't done fully correctly.

Reply:
Updated the code according to Yi's suggestion. japi-compliance-checker is still good.